// mongoose-auth/lib/modules/google/plugin.js
var mongoose = require('mongoose'), 
    mongooseTypes = require('mongoose-types'),
    _schema = require('./schema'),
    everyauth = require('everyauth');
mongooseTypes.loadTypes(mongoose);
var Email = mongoose.SchemaTypes.Email;

module.exports = function google(schema, opts) {
  schema.add(_schema);

  schema.static('createWithGoogle', function(googleUser, accessToken, otherAccessToken, callback) {

    var params = {
      google : {
        id : googleUser.id,
        name : {
          first : googleUser.given_name,
          last : googleUser.family_name
        },
        link : googleUser.link,
        picture : googleUser.picture,
        gender : googleUser.gender,
        locale : googleUser.locale,
        expiresIn : googleUser.expiresIn,
        accessToken : accessToken
      }
    };

    if (everyauth.password)
      params[everyauth.password.loginKey()] = "google:" + googleUser.id; // Hack because of way mongodb treate unique indexes

    this.create(params, callback);
  });
};
